src/jit/codegenxarch.cpp

```diff
@@ -3563,7 +3579,7 @@ void CodeGen::genStructPutArgUnroll(GenTreePutArgStk* putArgNode, unsigned baseV
     noway_assert(putArgNode->TypeGet() == TYP_STRUCT);
```

Comment:
IMO, We should add a comment on all checks whether we want to exclude SIMD types and explicitly check for TYP_STRUCT to avoid any confusion.  Because it is the easiest mistake to make to use

== TYP_STRUCT instead of varTypeIsStruct()